pkg/exchange/coinbase/active_order.go

```diff
@@ -10,20 +12,103 @@ import (
 type ActiveOrder struct {
 	submitOrder types.SubmitOrder
 	rawOrder    *api.CreateOrderResponse
+
+	lastUpdate time.Time
+}
+
+type registryKey struct {
+	key, secret, passphrase string
 }
 
+var actStoreRegistry map[registryKey]*ActiveOrderStore = make(map[registryKey]*ActiveOrderStore)
+
 type ActiveOrderStore struct {
 	mu     sync.Mutex
 	orders map[string]*ActiveOrder
+
+	startTime time.Time
+	ctx       context.Context
+	cancel    context.CancelFunc
 }
 
-func newActiveOrderStore() *ActiveOrderStore {
-	return &ActiveOrderStore{
+func newActiveOrderStore(key, secret, passphrase string) *ActiveOrderStore {
+	rk := registryKey{key, secret, passphrase}
+	if store, ok := actStoreRegistry[rk]; ok {
+		return store
+	}
+	store := &ActiveOrderStore{
 		orders: make(map[string]*ActiveOrder),
 	}
+	store.ctx, store.cancel = context.WithCancel(context.Background())
+	actStoreRegistry[rk] = store
+	return store
+}
+
+func (a *ActiveOrderStore) IsStarted() bool {
+	a.mu.Lock()
+	defer a.mu.Unlock()
+
+	return !a.startTime.IsZero()
 }
 
-func (a *ActiveOrderStore) getByUUID(orderUUID string) (*ActiveOrder, bool) {
+// Start starts the active order store cleanup worker.
+func (a *ActiveOrderStore) Start() {
+	a.mu.Lock()
+	defer a.mu.Unlock()
+
+	if a.startTime.IsZero() {
+		a.startTime = time.Now()
+		go a.cleanupWorker(a.ctx)
+	}
+}
+
+// Stop stops the active order store cleanup worker and resets the store.
+func (a *ActiveOrderStore) Stop() {
+	a.mu.Lock()
+	defer a.mu.Unlock()
+
+	if a.startTime.IsZero() {
+		return
+	}
+	a.cancel()                                                 // stop the cleanup worker
+	a.startTime = time.Time{}                                  // reset start time
+	a.orders = make(map[string]*ActiveOrder)                   // clear orders
+	a.ctx, a.cancel = context.WithCancel(context.Background()) // create new context
+}
+
+func (a *ActiveOrderStore) cleanupWorker(ctx context.Context) {
+	ticker := time.NewTicker(time.Minute * 5)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			coinbaseLogger.Info("active order store cleanup worker stopped")
+			return
+		case <-ticker.C:
+			a.mu.Lock()
+			for orderUUID, activeOrder := range a.orders {
```

Comment:
pull out this loop to a method, so that you can test it without allocating a ticker